src/services/shims.ts

```diff
@@ -322,8 +323,19 @@ module ts {
         }
     }
 
+    function serializeResultAsJson(target: Shim, key: string | symbol, descriptor: PropertyDescriptor) {

```

Comment:
this is a good idea,, pinging @rbuckton. so for instance we would define PropertyDecorator  as
```ts
declare type PropertyDecorator = <TKey extends string|symbol>(target: Object, propertyKey: TKey) => void;
```
and always instantiate it with either string or symbol based on the property kind when we are doing the comparison to issue the error..